command/root.go

```diff
@@ -59,21 +63,22 @@ func init() {
 		if err == pflag.ErrHelp {
 			return err
 		}
-		return &FlagError{Err: err}
+		return &cmdutil.FlagError{Err: err}
 	})
-}
-
-// FlagError is the kind of error raised in flag processing
-type FlagError struct {
-	Err error
-}
-
-func (fe FlagError) Error() string {
-	return fe.Err.Error()
-}
 
-func (fe FlagError) Unwrap() error {
-	return fe.Err
+	// TODO: iron out how a factory incorporates context
```

Comment:
It is indeed important, thanks for bringing it up. The api command was not really appropriate to explore more aspects of reading context in because it's so isolated. I agree that we should have a clear migration plan for other commands and, after merging this, what I plan to do is migrate a single older command over to the new approach and have that migration serve as a guidebook on how to apply this pattern.